src/core/platform.js

```diff
@@ -17,8 +17,10 @@ const detectPassiveEvents = () => {
 };
 
 const ua = (typeof navigator !== 'undefined') ? navigator.userAgent : '';
-const environment = typeof window !== 'undefined' ? 'browser' :
-    typeof global !== 'undefined' ? 'node' : 'worker';
+const environment =
+    typeof window !== 'undefined' ? 'browser' :
+        typeof WorkerGlobalScope !== 'undefined' ? 'webworker' :
+            typeof global !== 'undefined' ? 'node' : 'worker';
```

Comment:
So.....what's the difference between 'worker' and 'webworker'?

Reply:
`worker` was defined in #6108 by @marklundin:

> When a user script is parsed in the editor, it's executed in a Worker which doesn't have window or global.

When reading this, it was not clear to me that a Web Worker was meant by "Worker"; it's also coded as a fallback, and not coded explicitly like with `typeof WorkerGlobalScope`. So to be explicit, I introduced `webworker`. But, reading the above definition again, yes, it sounds like a the two are the same and can be unified. If there are no objections, I'll unify them.

In any case, #6108 only introduces `worker` but does not do anything with that value. It looks like #6108 would not have been created if this PR here was already merged.

Reply:
@willeastcott Please re-review.